operator/docs/enhancements/zone_aware_replication.md

```diff
@@ -0,0 +1,362 @@
+
+* Configure the availability zone for each ingester via the `-ingester.availability-zone` CLI flag (or its respective YAML config option)
+* Rollout ingesters to apply the configured zone
+* Enable time-series zone-aware replication via the `-distributor.zone-awareness-enabled` CLI flag (or its respective YAML config option). Please be aware this configuration option should be set to `distributors`, `queriers` and `rulers`.
+https://github.com/grafana/rollout-operator
+
+<div id="sizing"/>
+
+## Impact on LokiStack Sizing Configurations
+Without zone-aware replication, the LokiStack pods are scheduled on different nodes within the same or different availability zones. There are no hard requirements on the number of zones within the cluster. In case a zone fails, the pods are simply rescheduled to other nodes based on the current policies (taints/tolerations/affinity).
+
+If there are more than `floor(replication_factor/2)` zones with failing instances, reads and writes would not be possible and there would be a high chance of data loss. According to [Cortex](https://cortexmetrics.io/docs/guides/zone-aware-replication/#minimum-number-of-zones), the minimum number of zones should be equal to the replication factor. Which means the following for our production t-shirt sizes:
+* `1x.small` has a replication factor of 2 & all components have 2 replicas. In order to replicate a `1x.small` LokiStack across zones, there has to be at least 2 zones available in the cluster. Each replica of the LokiStack pods will be scheduled on a node in a different zone. For example: zone-a and zone-b will have 1 replica each of `distributor`, `ingester`, `querier`, `query-frontend`, `gateway`, `index-gateway`, and `ruler`.
+
+  The issue arises when we only have 2 zones, when one of them fails, the write-path cannot hold on if data ingestion rate is high, in turn data loss is possible. This can be overcome by only enabling zone-aware replication if the number of existing zones are `replication_factor + 1` (3 in this case, which is also the default for Cortex implementations). This solution is possible since most public cloud provider have 3 availability zones per region.
+
+  Switching a `1x.small` LokiStack to being zone-aware: setting `replication.topology.enabled` in the LokiStack CR to `true` should be sufficient to restart the pods and reschedule them on different zones (given that the requirements are met).
+
+* `1x.medium` has a replication factor of 3, with the `ingester` and `querier` having 3 replicas, while the `distributor`, `query-frontend`, `gateway`, `index-gateway`, and `ruler` having only 2 replicas. This means 1 zone will have only 1 `querier` and 1 `ingester` that will communicate with components (e.g. distributors) in other zones . In this case, losing one zone cannot be tolerated, and relying on having a 4th zone isn't always possible. In this case it is preferable to suggest a new sizing configuration.
```

Comment:
```suggestion
* `1x.medium` has a replication factor of 3, with the `ingester` and `querier` having 3 replicas, while the `distributor`, `query-frontend`, `gateway`, `index-gateway`, and `ruler` having only 2 replicas. This means 1 zone will have only 1 `querier` and 1 `ingester` that will communicate with components (e.g. distributors) in other zones . In this case, losing one zone cannot be tolerated, and relying on having a 4th zone isn't always possible. In this case it is preferable to suggest using a replication factor of 2 instead of the default set to 3.
```